src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -227,9 +233,25 @@ private function getDocBlockFromMethod(string $class, string $ucFirstProperty, i
         }
 
         try {
-            return [$this->docBlockFactory->create($reflectionMethod, $this->contextFactory->createFromReflector($reflectionMethod)), $prefix];
+            return [$this->docBlockFactory->create($reflectionMethod, $this->createFromReflector($reflectionMethod->getDeclaringClass())), $prefix];
         } catch (\InvalidArgumentException $e) {
             return null;
         }
     }
+
+    /**
+     * Prevents a lot of redundant calls to ContextFactory::createForNamespace().
```

Comment:
`ContextFactory::createFromReflector()`?

Reply:
this is indeed the case but what I really want to avoid is calls to `createForNamespace`. But this is private and the class itself is final, so...

This is the intention, but I can of course update the comment.

Reply:
What I find a bit dangerous is that our in-memory cache now depends on some assumptions on the inner-working of the phpDocumentor package (the comment here being one of the assumptions). That means that we need to closely  follow updates made to a third-party package to be sure we do not break anything here.

Reply:
And there is no other way, right?

Reply:
We could also add this catch directly in `phpDocumentor/TypeResolver` isn't it?

Reply:
@dunglas I thought so, but this is a utilitary class, it should not increase the memory used mandatory. It could eventually have another public method that handle this cache, but it seems to me that it's more logical to decorate it by ourself, it symfony after all that chose to not use the data once and calls it multiple times.